reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -6602,17 +6637,42 @@ public int getPrefetch() {
 	 * @see #first
 	 */
 	public final Flux<T> or(Publisher<? extends T> other) {
-		if (this instanceof FluxFirstEmitting) {
-			FluxFirstEmitting<T> publisherAmb = (FluxFirstEmitting<T>) this;
+		if (this instanceof FluxFirst) {
+			FluxFirst<T> orPublisher = (FluxFirst<T>) this;
 
-			FluxFirstEmitting<T> result = publisherAmb.ambAdditionalSource(other);
+			FluxFirst<T> result = orPublisher.orAdditionalSource(other);
 			if (result != null) {
 				return result;
 			}
 		}
 		return first(this, other);
 	}
 
+	/**
+	 * Pick the first {@link Publisher} between this {@link Flux} and another publisher
+	 * to emit any value and replay all signals from that
+	 * {@link Publisher}, effectively behaving like the fastest of these competing sources.
+	 *
+	 * <p> TODO
+	 * <img class="marble" src="doc-files/marbles/orForFlux.svg" alt="">
+	 *
+	 * @param other the {@link Publisher} to race with
+	 *
+	 * @return the fastest sequence
+	 * @see #firstValues
+	 */
+	public final Flux<T> orValues(Publisher<? extends T> other) {
+		if (this instanceof FluxFirstValues) {
+			FluxFirstValues<T> orPublisher = (FluxFirstValues<T>) this;
+
+			FluxFirstValues<T> result = orPublisher.orAdditionalSource(other);
+			if (result != null) {
+				return result;
+			}
+		}
+		return firstValues(this, other);
+	}
```

Comment:
After further thoughts, I'm having huge doubts about the introduction of this method and the corresponding one in Mono as I think it can easily be missed by the user and though results in an invalid mix & match of firstValues and or operators called.

Reply:
that is actually a huge argument in favor of NOT mutualizing the publisher concrete classes !